python/cudf/cudf/comm/serialize.py

```diff
@@ -40,9 +40,10 @@ def dask_serialize_cudf_object(x):
     def deserialize_cudf_object(header, frames):
         with log_errors():
             if header["serializer"] == "cuda":
-                assert all(
-                    hasattr(f, "__cuda_array_interface__") for f in frames
-                )
+                for f in frames:
+                    # some frames are empty -- meta/empty partitions/etc
+                    if len(f) > 0:
+                        assert hasattr(f, "__cuda_array_interface__")
```

Comment:
Does something happen upstream in the case of a 0 length buffer? Our Buffer objects should still have a `__cuda_array_interface__` even if they're 0 length.

Reply:
That is a good question.  I am not sure how this is generate

Reply:
I think we may be using `bytes(0)` or similar as a filler. They are special cased so as to not actually be sent.

Reply:
Relevant [send code]( https://github.com/dask/distributed/blob/384080422f5bd54d16ba23161a1c2d18f74ff299/distributed/comm/ucx.py#L164-L165 ) and [receive code]( https://github.com/dask/distributed/blob/384080422f5bd54d16ba23161a1c2d18f74ff299/distributed/comm/ucx.py#L202-L205 ).

Though I guess we do [create a `DeviceBuffer(size=0)`]( https://github.com/dask/distributed/blob/384080422f5bd54d16ba23161a1c2d18f74ff299/distributed/comm/ucx.py#L203 ). So maybe this check is ok?

Where did you run into this @quasiben?